### PR TITLE
Main

### PR DESCRIPTION
**Pull request by MarynaHrushevska**

### CODE REVIEW

File: src/components/App/styled.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { createGlobalStyle } from 'styled-components';
 import { IStyled } from 'common/types/styledTypes';
 
-export const StGlobalStyle = createGlobalStyle <IStyled>`
+export const StGlobalStyle = createGlobalStyle < IStyled > `
```
Comment (ghost) [inline src/components/App/styled.ts]: fix in all PR

File: src/components/Checkers/styled.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+export const StCell = styled.div < IStyled > `
+  width: 80px;
+  height: 80px;
+  border: 1px solid ${({ colors, theme }) => colors[theme].textColor};
+  margin: 1px;
+  color: ${({ colors, theme }) => colors[theme].textColor};
+  background: #ccc;
+  transform: rotate(180deg);
+  
+  &:nth-child(2),
```
Comment (ghost) [inline src/components/Checkers/styled.ts:28]: research another approach

File: src/components/Header/HeaderLanguage/HeaderLanguage.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React, { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useTheme } from 'src/components/hooks/useTheme';
+import { StLangToggle } from './styled';
+
+const HeaderLanguage: React.FC = () => {
+  const [lang, setLang] = useState(localStorage.getItem('lang') || 'en');
```
Comment (ghost) [inline src/components/Header/HeaderLanguage/HeaderLanguage.tsx]: type for useState

File: src/components/Header/HeaderTheme/HeaderTheme.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { StThemeToggle } from './styled';
+import { useTheme } from '../../hooks/useTheme';
+
+interface IProps {
+  toggleThemeMode: () => void;
+}
+
+const HeaderTheme: React.FC <IProps> = ({ toggleThemeMode }: IProps) => {
```
Comment (ghost) [inline src/components/Header/HeaderTheme/HeaderTheme.tsx]: remove redundant IProps

File: src/components/Main/MainRoomsList/MainRoomsList.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import { IRoom } from 'store/room/types';
+import { StP, StRooms, StRoomsContainer } from './styled';
+import MainRoomsItem from '../MainRoomsItem';
+import AddRoomBtn from '../AddRoomBtn';
+
+interface IProps {
+    rooms: IRoom[],
+}
+
+const MainRoomsList: React.FC <IProps> = ({ rooms }: IProps) => {
```
Comment (ghost) [inline src/components/Main/MainRoomsList/MainRoomsList.tsx]: same in all project remove redundant IProps in the end

File: src/components/Header/HeaderLanguage/HeaderLanguage.tsx (hunk 3 above)
Comment (MarynaHrushevska) [reply src/components/Header/HeaderLanguage/HeaderLanguage.tsx]: changes are in branch 'socket'

File: src/components/Checkers/styled.ts (hunk 2 above)
Comment (MarynaHrushevska) [reply src/components/Checkers/styled.ts:28]: changes are in branch 'socket'

File: src/components/Checkers/Checkers.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+              <StCell
+                  key={cell.id}
+                  theme={theme}
+                  colors={colors}
+              >
+                  {
+                      (cell.hasItem)
+                        ?
+                          (cell.color === 'black')
+                          ? <StCellBlack colors={colors} theme={theme} draggable/>
```
Comment (ghost) [inline src/components/Checkers/Checkers.tsx:25]: black to const

File: src/components/Checkers/styled.ts [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+  background: ${({ colors, theme }) => colors[theme].backgroundSecondary};
+  transform: rotate(180deg);
+`;
+export const StCell = styled.div <IStyled>`
+  width: 80px;
+  height: 80px;
+  border: 1px solid ${({ colors, theme }) => colors[theme].textColor};
+  margin: 1px;
+  color: ${({ colors, theme }) => colors[theme].textColor};
+  background: #ccc;
```
Comment (ghost) [inline src/components/Checkers/styled.ts:25]: no hardcode for colors

File: src/components/Checkers/styled.ts [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  &:nth-child(50),
+  &:nth-child(52),
+  &:nth-child(54),
+  &:nth-child(56),
+  &:nth-child(57),
+  &:nth-child(59),
+  &:nth-child(61),
+  &:nth-child(63)
+    {
+    background: #fff;
```
Comment (ghost) [inline src/components/Checkers/styled.ts:61]: !

File: src/components/Checkers/styled.ts [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+      left: 50%;
+      transform: translate(-50%, -50%);
+      width: 25px;
+      height: 25px;
+      border-radius: 50%;
+    }
+  }
+`;
+export const StCellWhite = styled.div <IStyled>`
+  border: 2px solid #000;
```
Comment (ghost) [inline src/components/Checkers/styled.ts:87]: check everywhere and replace to theme

File: src/components/Header/Header.tsx [hunk 10]
```diff
@@ -1,11 +1,47 @@
...
+import { Theme } from '../hocs';
+import HeaderTheme from './HeaderTheme';
+import HeaderLanguage from './HeaderLanguage';
+import HeaderRoute from './HeaderRoute';
+import HeaderLogout from './HeaderLogout';
+
+const Header: React.FC = ({ location }) => {
+    const { theme, changeTheme } = useContext(Theme);
+    const toggleThemeMode = () => {
+        changeTheme(theme === 'dark' ? 'light' : 'dark');
```
Comment (ghost) [inline src/components/Header/Header.tsx:13]: dark, light to const

File: src/components/Header/styled.ts [hunk 11]
```diff
@@ -1,12 +1,24 @@
 import styled from 'styled-components';
 
 export const StHeader = styled.div`
+    display: flex;
+    justify-content: space-around;
+    align-items: center;
     width: 100vw;
     height: 90px;
     background: #2B5278;
```
Comment (ghost) [inline src/components/Header/styled.ts:9]: !

File: src/components/Main/MainRoomsList/MainRoomsList.tsx [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+            <StRoomsContainer>
+                {rooms.length 
+                    ? rooms.map((room) => (
+                        <MainRoomsItem 
+                            key={room.id} 
+                            type={room.gameType}
+                            login={room.loginName} 
+                        />
+                    ))  
+                    : <StP>No rooms</StP>}
```
Comment (ghost) [inline src/components/Main/MainRoomsList/MainRoomsList.tsx:31]: use translation

File: src/components/Modal/ModalButtons/ModalButtons.tsx [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+        user,
+        addRoom,
+        gameType,
+        handleCloseModal
+    }) => {
+    const { t } = useTranslation();
+    const { colors, theme } = useTheme();
+    const handleCreateRoom = () => {
+        const obj: IRoom = {
+            id: Math.random() * 1000,
```
Comment (ghost) [inline src/components/Modal/ModalButtons/ModalButtons.tsx:29]: move logic to helper create uniq uuid investigate how you can do it use es6 features

File: src/components/Modal/ModalCross/ModalCross.tsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+
+const ModalCross: React.FC <IProps> = ({ handleCloseModal }) => {
+    const { colors, theme } = useTheme();
+    return (
+        <StCrossBtn 
+            theme={theme}
+            colors={colors}
+            onClick={handleCloseModal}
+        >
+            x
```
Comment (ghost) [inline src/components/Modal/ModalCross/ModalCross.tsx:17]: to const

File: src/components/Room/Room.tsx [hunk 15]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import Checkers from 'src/components/Checkers';
+
+const Room = () => {
+    return (
```
Comment (ghost) [inline src/components/Room/Room.tsx:5]: return

File: src/components/Statistic/Statistic.tsx [hunk 16]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+
+const Statistic = () => {
+  return (
```
Comment (ghost) [inline src/components/Statistic/Statistic.tsx:4]: return

File: src/components/Statistic/styled.ts [hunk 17]
```diff
@@ -0,0 +1,8 @@
+import styled from 'styled-components';
+
+export const StContainer = styled.div`
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  color: #fff;
```
Comment (ghost) [inline src/components/Statistic/styled.ts:7]: !

**Pull request closed without merging**